### PR TITLE
feat: migrate mvc dispatcher from /api to /

### PR DESCRIPTION
**Pull request by netroms**
## Summary
This PR moves the default dispatcher servlet mapping from `/api` to `/`.
This change affected all the controllers `@RequestMapping`, hence all the controllers have now the `/api`, prefixed to the endpoint string. While doing this refactoring, I also made sure when endpoint strings was referred from other classes or variables, it was moved into the actual RequestMappings's value for convince.

As a side effect of this move, we can no longer serve static content from the root of the application since everything from `/*` is mapped to the main dispatcher servlet, so we now serve static content (apps and resources) with Spring's `ResourceHandler`, configured in the `WebMvcConfig` class.

We no longer include the build .war of the apps module and the static resource module. Instead, the portal and the embedded Jetty, reads from the already build module's target folder and copies the content into the final .war file (for the portal module), for the embedded Jetty module we read the resources directly from the filesystem with the same Spring `ResourceHandler`.

This PR is an intermediate step before the embedded Tomcat JIRA, as to not make this one too big: [DHIS2-14977](https://dhis2.atlassian.net/browse/DHIS2-14977)

[DHIS2-14977]: https://dhis2.atlassian.net/browse/DHIS2-14977?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/security/config/WebMvcConfig.java [hunk 1]
```diff
@@ -134,6 +139,66 @@ public class WebMvcConfig extends DelegatingWebMvcConfiguration {
...
+  //        HttpServletRequest request, Resource resource, ResourceTransformerChain
+  // transformerChain)
+  //        throws IOException {
+  //      resource = transformerChain.transform(request, resource);
+  //      if (!"fileExtension".equals(StringUtils.getFilenameExtension(resource.getFilename()))) {
+  //        return resource;
+  //      }
+  //      return resource;
+  //    }
+  //  }
```
Comment (jbee) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/security/config/WebMvcConfig.java:183]: remove?

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/security/config/WebMvcConfig.java [hunk 2]
```diff
@@ -134,6 +139,66 @@ public class WebMvcConfig extends DelegatingWebMvcConfiguration {
...
+    registry
+        .setOrder(Ordered.LOWEST_PRECEDENCE)
+        .addResourceHandler("/**")
+        .addResourceLocations(
+            "/resources/",
+            "/dhis/",
+            "classpath:/static/",
+            "file:./dhis-web-apps/target/dhis-web-apps/",
+            "file:./dhis-web-commons-resources/src/main/webapp/")
+        // .setCachePeriod(3600)
```
Comment (jbee) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/security/config/WebMvcConfig.java:196]: remove commented out lines?

File: dhis-2/dhis-web-embedded-jetty/src/main/java/org/hisp/dhis/web/embeddedjetty/EmbeddedJettyBase.java [hunk 3]
```diff
@@ -73,19 +70,21 @@ public void startJetty() throws Exception {
...
+    //    RedirectPatternRule rewritePatternRule = new RedirectPatternRule();
+    //    rewritePatternRule.setPattern("");
+    //    rewritePatternRule.setLocation("/dhis-web-login");
+    //    rewrite.addRule(rewritePatternRule);
 
     HandlerList handlers = new HandlerList();
-    handlers.setHandlers(new Handler[] {rewrite, getServletContextHandler(), new DefaultHandler()});
+    //    handlers.setHandlers(new Handler[] {rewrite, getServletContextHandler(), new
+    // DefaultHandler()});
+    handlers.setHandlers(new Handler[] {getServletContextHandler(), new DefaultHandler()});
```
Comment (jbee) [inline dhis-2/dhis-web-embedded-jetty/src/main/java/org/hisp/dhis/web/embeddedjetty/EmbeddedJettyBase.java:87]: remove commented out lines?

**Review by jbee — COMMENTED** [review]
Looks good!

A few places that need cleanup of commented out code.
In one controller I think the mapping got wrong having the `/api` part both on the class and on the methods

**Review by jbee — APPROVED** [review]

**Review by vietnguyen — APPROVED** [review]

**Pull request merged**